src/core/blockchain/ethereum/client.ts

```diff
@@ -111,6 +147,143 @@ export class Client extends BlockchainGenericClient {
+                    rpcCalls.push(
+                        this.estimateGas(
+                            false,
+                            data.from,
+                            data.to,
+                            data.contractAddress,
+                            new BigNumber(data.amount),
+                            data.raw
+                        )
+                    );
+                    break;
+                }
+            }
+            let presets: {
+                low: {
+                    maxFeePerGas: BigNumber;
+                    maxPriorityFeePerGas: BigNumber;
+                };
+                medium: {
+                    maxFeePerGas: BigNumber;
+                    maxPriorityFeePerGas: BigNumber;
+                };
+                high: {
+                    maxFeePerGas: BigNumber;
+                    maxPriorityFeePerGas: BigNumber;
+                };
+            };
+            rpcCalls.push(this.getBaseFeeHistory(), this.getMaxPriorityFee());
+
+            const results = await Promise.all(rpcCalls);
+            if (results[0] && results[1] && results[2]) {
+                presets = {
+                    low: {
+                        maxFeePerGas: new BigNumber(results[1]).plus(results[2]),
+                        maxPriorityFeePerGas: new BigNumber(results[2])
+                    },
+
+                    medium: {
+                        maxFeePerGas: new BigNumber(results[1]).multipliedBy(1.5).plus(results[2]),
+                        maxPriorityFeePerGas: new BigNumber(results[2]).multipliedBy(1.5)
+                    },
+                    high: {
+                        maxFeePerGas: new BigNumber(results[1]).multipliedBy(2).plus(results[2]),
+                        maxPriorityFeePerGas: new BigNumber(results[2]).multipliedBy(2)
+                    }
+                };
+            }
+
+            const gasLimit =
+                results[0] && results[0].result
+                    ? new BigNumber(parseInt(results[0].result, 16))
+                    : config.feeOptions.defaults.gasLimit[tokenType];
+
+            return {
+                maxFeePerGas: results[1].toString(),
+                maxPriorityFeePerGas: results[2].toString(),
+                gasLimit: gasLimit.toString(),
+                presets: presets ? presets : config.feeOptions.defaults.gasPricePresets,
+                feeTotal: new BigNumber(results[1])
+                    .multipliedBy(2)
+                    .multipliedBy(gasLimit)
+                    .toString(),
+                responseHasDefaults: presets ? false : true
+            };
+        } catch (error) {
+            const gasPrice = config.feeOptions.defaults.gasPrice;
+            const gasLimit = config.feeOptions.defaults.gasLimit[tokenType];
+
+            SentryCaptureException(
+                new Error(JSON.stringify({ event: 'getEstimated Fees - defaults Set', error }))
```

Comment:
You can use this pattern. Send the error in Breadcrumb and keep the error message short and concise.

``` 
      SentryAddBreadcrumb({
            message: JSON.stringify({
                error
            })
        });
        SentryCaptureException(
            new Error(`Failed to get estimated fees - defaults Set, ${error?.message}`)
        );
```